crates/ruff_linter/src/rules/pylint/rules/magic_value_comparison.rs

```diff
@@ -51,42 +51,37 @@ impl Violation for MagicValueComparison {
     }
 }
 
-/// If an [`Expr`] is a constant (or unary operation on a constant), return the [`Constant`].
-fn as_constant(expr: &Expr) -> Option<&Constant> {
+/// If an [`Expr`] is a literal (or unary operation on a literal), return the literal [`Expr`].
+fn as_literal(expr: &Expr) -> Option<&Expr> {
```

Comment:
Another potential use for a virtual enum for literal expression.

Reply:
What's stopping you from adding one?

I think it would also be nice to have a `Expr::as_literal_expr()` method for the situations where we want to handle all literal types.